crates/bevy_scene/src/dynamic_scene_builder.rs

```diff
@@ -274,11 +274,22 @@ impl<'w> DynamicSceneBuilder<'w> {
                         return None;
                     }
 
-                    let component = type_registry
-                        .get(type_id)?
+                    let type_registration = type_registry.get(type_id)?;
+
+                    let component = type_registration
                         .data::<ReflectComponent>()?
                         .reflect(original_entity)?;
-                    entry.components.push(component.clone_value());
+
+                    // Clone the via `FromReflect`. Unlike `PartialReflect::clone_value` this
+                    // retains the original type and `ReflectSerialize` type data which is needed to
+                    // deserialize.
```

Comment:
Nit: Could we add a test to verify this works?

Reply:
Will do.

Reply:
Added. I verified the test works without the fix.